ranger/config/commands.py

```diff
@@ -1959,3 +1959,46 @@ def tab(self, tabnum):
             in sorted(self.modes.keys())
             if mode
         )
+
+
+class confirm(Command):
+    """:confirm <command_to_be_confirmed> + <potential command arguments>
+
+        Can be used with any command. Ranger will ask if it should execute the
+        given command.
+    """
+
+    def execute(self):
+        from functools import partial
+
+        command = ''
+        if len(self.args) > 1 and self.arg(1) != 'chain':
+            for index in range(1, len(self.args)):
+                if index != len(self.args):
+                    command += self.args[index] + ' '
+            self.fm.ui.console.ask(
+                "Confirm execution of: %s (y/N)" % ''.join(command),
+                partial(self._question_callback, command),
+                ('n', 'N', 'y', 'Y'),
+            )
+        else:
+            arguments = self.rest(2).split(';')
+            for index, item in enumerate(arguments):
+                arguments[index] = item.strip()
+            self._chaining(len(arguments) - 1, arguments)
+
+    def _question_callback(self, command, answer):
+        if answer == 'y' or answer == 'Y':
+            self.fm.execute_console(command)
+
+    def _chaining(self, recursion_count, commands):
+        from functools import partial
+        if recursion_count >= 0:
```

Comment:
Why do we need this count?

Reply:
Well I could work with len() of the list, or delete a command from the list after every recursion and check if the len() is still greater 0

Reply:
Aren't you already splitting the list on `;`s?

Reply:
Yes but each command is one element of a list . I used the counter to determine how long the recursion should go on. When the counter reaches zero, the first element of the commands list will be called (commands will be called from last to first, I could reverse that order if preferred), on the next iteration the counter will be -1 and the recursion stops. To me it seemed pretty straight forward.